src/test/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/dateinput/DateValidationSupportSpec.scala

```diff
@@ -0,0 +1,306 @@
+    "date is valid" should {
+
+      "convert valid date with various months to LocalDate" should {
+        val months = Table(
+          ("month", "expectedMonth"),
+          // English month names
+          ("January", 1),
+          ("February", 2),
+          ("March", 3),
+          ("April", 4),
+          ("May", 5),
+          ("June", 6),
+          ("July", 7),
+          ("August", 8),
+          ("September", 9),
+          ("October", 10),
+          ("November", 11),
+          ("December", 12),
+          // English abbreviated month names, where different
+          ("Jan", 1),
+          ("Feb", 2),
+          ("Mar", 3),
+          ("Apr", 4),
+          ("Jun", 6),
+          ("Jul", 7),
+          ("Aug", 8),
+          ("Sep", 9),
+          ("Oct", 10),
+          ("Nov", 11),
+          ("Dec", 12),
+          ("3", 3),
+          // Welsh month names
+          ("Ionawr", 1),
+          ("Chwefror", 2),
+          ("Mawrth", 3),
+          ("Ebrill", 4),
+          ("Mai", 5),
+          ("Mehefin", 6),
+          ("Gorffennaf", 7),
+          ("Awst", 8),
+          ("Medi", 9),
+          ("Hydref", 10),
+          ("Tachwedd", 11),
+          ("Rhagfyr", 12),
+          // Welsh abbreviated month names, where different (according to the library we're using)
+          ("Ion", 1),
+          ("Chwef", 2),
+          ("Maw", 3),
+          ("Meh", 6),
+          ("Gorff", 7),
+          ("Hyd", 10),
+          ("Tach", 11),
+          ("Rhag", 12)
+        )
+
+        forAll(months) { (enteredMonth: String, expectedMonth: Int) =>
+          s"work for month $enteredMonth" in new Setup {
+            val day                          = "1"
+            val year                         = "2023"
+            val form: Form[SomeFormWithDate] = testForm.bind(formData(day, enteredMonth, year))
+            form.errors should be(Nil)
+            form.value  should be(
+              Some(
+                SomeFormWithDate(
+                  GovUkDate(
+                    DateEntered(day, MonthEntered(enteredMonth, expectedMonth), year),
+                    LocalDate.of(year.toInt, expectedMonth, day.toInt)
+                  )
+                )
+              )
+            )
+          }
+        }
+      }
+
+      "convert valid date with numeric month to LocalDate" in new Setup {
+        val form: Form[SomeFormWithDate] = testForm.bind(formData("1", "12", "2023"))
+        form.errors should be(Nil)
+        form.value  should be(
+          Some(SomeFormWithDate(GovUkDate(DateEntered("1", MonthEntered("12", 12), "2023"), LocalDate.of(2023, 12, 1))))
+        )
+      }
+    }
+
+    "day is invalid or missing" should {
+      "reject invalid date where day < 1" in new Setup {
+        val form: Form[SomeFormWithDate] = testForm.bind(formData("0", "12", "2023"))
+        form.errors should be(
+          List(FormError("dateOfBirth.day", Seq("govuk.dateInput.error.date.invalid"), Seq("Date")))
```

Comment:
I kind of wonder - stepping back, if something useful from us - rather than an implementation, could be a table of test data that shows what error would be expected on which field - so "how to interpret and verify that you've met the current govuk guidance" that could then lead on (possibly) to a formatter that supports it - but maybe to calculate how that should look we'd want some input from service devs

I think I favour at the moment if we provide an example implementation, making it as simple as possible - so for just have a Formatter[LocalDate] and not (yet) doing extra stuff like capturing the entered values

It seems like people could still capture the month entered themselves if they did want to run away with it, for example

```scala
val whatIsYourDateOfBirthForm = Form(tuple(
  "dateOfBirth"       -> govukDate.verifying("error.youAreTooOld", _.isAfter(theMillennium)),
  "dateOfBirth.month" -> text
))

// user has gone back to edit something
val formFilledFromSomeDataStore = whatIsYourDateOfBirthForm.fill((
  LocalDate.of(2023, 10, 19),
  "Oct"
))

formFilledFromSomeDataStore("dateOfBirth.month").value // => Some("Oct")
```

and the reason being, that I think a `Formatter[LocalDate]` might not be exactly what we want in the end, but it feels like the jump from something that validates and parses a date implementing the govuk conventions to something that's optimally ergonomic for hmrc service devs feels like it would be better to do with input from service devs (which they might turn around and say _"no thanks, I will use your example as a reference of how to parse date, but I'm going to keep using my own implementation / I will just copy your one and then modify it in my service rather than use yours directly"_)